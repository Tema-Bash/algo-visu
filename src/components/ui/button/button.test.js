import { fireEvent, render } from '@testing-library/react'
import Button from './button.tsx'

describe('button tests', () => {
  it('button with text check', () => {
    const buttonString = 'num'
    const container = render(<Button text={buttonString} />)
    const buttonText = container.getByText(buttonString)
    expect(buttonText.textContent).toEqual(buttonString)
    expect(container).toMatchSnapshot()
  })

  it('button without text check', () => {
    const buttonString = ''
    const container = render(<Button text={buttonString} />)
    const buttonText = container.getByTestId('buttonText')
    expect(buttonText.textContent).toEqual('')
    expect(container).toMatchSnapshot()
  })

  it('button disabled check', () => {
    const buttonString = 'textButton'
    const mockHandler = jest.fn()
    const container = render(
      <Button text={buttonString} disabled={true} onClick={mockHandler} />
    )
    const button = container.getByTestId('button')
    fireEvent.click(button)
    expect(mockHandler).toHaveBeenCalledTimes(0)
    expect(button).toHaveProperty(`disabled`, true)
    expect(button).toBeDisabled()
    expect(container).toMatchSnapshot()
  })

  it('button loader check', () => {
    const buttonString = 'textButton'
    const mockHandler = jest.fn()
    const container = render(
      <Button text={buttonString} isLoader={true} onClick={mockHandler} />
    )
    const button = container.getByTestId('button')
    fireEvent.click(button)
    expect(mockHandler).toHaveBeenCalledTimes(0)

    const loaderImage = container.getByTestId('imgLoader')
    expect(loaderImage).toHaveProperty(`alt`, 'Загрузка.')
    expect(container).toMatchSnapshot()
  })
})
