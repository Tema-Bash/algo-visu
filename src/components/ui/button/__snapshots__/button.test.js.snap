// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button tests button disabled check 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="text text_type_button text_color_primary button undefined false "
        data-testid="button"
        disabled=""
        type="button"
      >
        <p
          class="text undefined"
          data-testid="buttonText"
        >
          textButton
        </p>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text text_type_button text_color_primary button undefined false "
      data-testid="button"
      disabled=""
      type="button"
    >
      <p
        class="text undefined"
        data-testid="buttonText"
      >
        textButton
      </p>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`button tests button loader check 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="text text_type_button text_color_primary button undefined loader "
        data-testid="button"
        disabled=""
        type="button"
      >
        <img
          alt="Загрузка."
          class="loader_icon"
          data-testid="imgLoader"
          src="loader.svg"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text text_type_button text_color_primary button undefined loader "
      data-testid="button"
      disabled=""
      type="button"
    >
      <img
        alt="Загрузка."
        class="loader_icon"
        data-testid="imgLoader"
        src="loader.svg"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`button tests button with text check 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="text text_type_button text_color_primary button undefined false "
        data-testid="button"
        type="button"
      >
        <p
          class="text undefined"
          data-testid="buttonText"
        >
          num
        </p>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text text_type_button text_color_primary button undefined false "
      data-testid="button"
      type="button"
    >
      <p
        class="text undefined"
        data-testid="buttonText"
      >
        num
      </p>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`button tests button without text check 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="text text_type_button text_color_primary button undefined false "
        data-testid="button"
        type="button"
      >
        <p
          class="text undefined"
          data-testid="buttonText"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text text_type_button text_color_primary button undefined false "
      data-testid="button"
      type="button"
    >
      <p
        class="text undefined"
        data-testid="buttonText"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
